python/lsst/ts/watcher/rules/mtm2_mirror_temperature.py

```diff
@@ -0,0 +1,153 @@
+        schema_yaml = """
+            $schema: 'http://json-schema.org/draft-07/schema#'
+            description: Configuration for MTM2MirrorTemperature rule.
+            type: object
+            properties:
+                ring:
+                    description: >-
+                        Limit of the ring temperature (degree C).
+                    type: number
+                    default: 25.0
+                intake:
+                    description: >-
+                        Limit of the intake (plenum) temperature (degree C).
+                    type: number
+                    default: 25.0
+                gradient:
+                    description: >-
+                        Limit of the gradient (or the variation of ring
+                        temperatures, peak-to-valley) temperature (degree C).
+                    type: number
+                    default: 10.0
+                poll_interval:
+                    description: Time delay between polling updates (second).
+                    type: number
+                    default: 1.0
+
+            required:
+            - ring
+            - intake
+            - gradient
+            - poll_interval
+            additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+
+    def setup(self, model) -> None:
+
+        self._remote = model.remotes[("MTM2", 0)]
+
+    def compute_alarm_severity(self) -> AlarmSeverityReasonType:
+        """Compute and set alarm severity and reason.
+
+        Returns
+        -------
+        None, if no change or unknown, or a tuple of two values:
+
+        severity: `lsst.ts.xml.enums.Watcher.AlarmSeverity`
+            The new alarm severity.
+        reason : `str`
+            Detailed reason for the severity, e.g. a string describing
+            what value is out of range, and what the range is.
+            If ``severity`` is ``NONE`` then this value is ignored (but still
+            required) and the old reason is retained until the alarm is reset
+            to ``nominal`` state.
+
+        Notes
+        -----
+        You may return `NoneNoReason` if the alarm state is ``NONE``.
+        """
+
+        assert self._remote is not None
+
+        sources = list()
+        if self._remote.tel_temperature.has_data:
+            data = self._remote.tel_temperature.get()
+
+            temperature_ring = np.array(data.ring)
+            if np.any(temperature_ring > self.config.ring):
+                sources.append("ring")
+
+            if (temperature_ring.max() - temperature_ring.min()) > self.config.gradient:
+                sources.append("gradient")
+
+            if np.any(np.array(data.intake) > self.config.intake):
+                sources.append("intake/plenum")
+
+        return (
+            NoneNoReason
+            if (len(sources) == 0)
+            else (
+                AlarmSeverity.WARNING,
+                f"Mirror temperature ({sources}) out of normal range.",
```

Comment:
I would suggest you make the "reason" a static string here. Consider a scenario where the "sources" out of range are constantly changing. This would make the alarm goes off at every change. If the reason is static, the alarm would continue to be active but would not be sounding as a new alarm. Once we know that the alarm is active, we can always go look for the sources in the EFD.

Reply:
Fixed in the commit: [e5e703b](https://github.com/lsst-ts/ts_watcher/pull/83/commits/e5e703b275b024b0fd34185ef4cae12882298a75). Thanks!